app/controller.js

```diff
@@ -75,6 +96,101 @@ class Controller {
       return res.status(500).json(error);
     }
   }
+
+  async forgot(req, res) {
+    // Send out an email with a token for password recovery
+    try {
+      const user = await User.findOne({ where: { email: req.body.email } });
+      if (!user) return res.status(404).json({ error: 'This email address is not in use.' });
+      const token = await crypto.randomBytes(20).toString('hex');
+      await cache.set(token, user.email);
+      const response = `${
+        environment === 'production' ? `https://${req.hostname}` : `http://localhost:${port}`
+      }/recover/${token}`;
+      sendMail(user.email, 'Password Recovery', response);
+      return res.status(200).json({ initiatedPasswordRecovery: true });
+    } catch (error) {
+      return res.status(500).json(error);
+    }
+  }
+
+  async recover(req, res) {
+    // Recover the password
+    try {
+      const email = cache.get(req.params.token);
+      if (email === undefined) return res.status(500).json({ error: 'An error has occured.' });
+      const user = await User.findOne({ where: { email } });
+      if (!user) return res.status(404).json({ error: 'This email address is not in use.' });
+      await user.update({ password: hashSync(req.body.password, 10) });
```

Comment:
I wonder if we should do more rounds for the password. Standard practice right now seems to be 12 or 14.